pkg/proxy/util/utils.go

```diff
@@ -255,26 +255,53 @@ func LogAndEmitIncorrectIPVersionEvent(recorder record.EventRecorder, fieldName,
 	}
 }
 
-// FilterIncorrectIPVersion filters out the incorrect IP version case from a slice of IP strings.
-func FilterIncorrectIPVersion(ipStrings []string, ipfamily v1.IPFamily) ([]string, []string) {
-	return filterWithCondition(ipStrings, (ipfamily == v1.IPv6Protocol), utilnet.IsIPv6String)
+// MapIPsByIPFamily maps a slice of IPs to their respective IP families (v4 or v6)
+func MapIPsByIPFamily(ipStrings []string) map[v1.IPFamily][]string {
+	ipFamilyMap := map[v1.IPFamily][]string{}
+	for _, ip := range ipStrings {
+		// Handle only the valid IPs
+		if net.ParseIP(ip) != nil {
+			ipFamily := getIPFamilyFromIP(ip)
+			ipFamilyMap[ipFamily] = append(ipFamilyMap[ipFamily], ip)
+		}
+	}
+	return ipFamilyMap
 }
 
-// FilterIncorrectCIDRVersion filters out the incorrect IP version case from a slice of CIDR strings.
-func FilterIncorrectCIDRVersion(ipStrings []string, ipfamily v1.IPFamily) ([]string, []string) {
-	return filterWithCondition(ipStrings, (ipfamily == v1.IPv6Protocol), utilnet.IsIPv6CIDRString)
+// MapCIDRsByIPFamily maps a slice of IPs to their respective IP families (v4 or v6)
+func MapCIDRsByIPFamily(ipStrings []string) map[v1.IPFamily][]string {
+	ipFamilyMap := map[v1.IPFamily][]string{}
+	for _, cidr := range ipStrings {
+		// Handle only the valid CIDRs
+		if _, _, err := net.ParseCIDR(cidr); err == nil {
+			ipFamily := getIPFamilyFromCIDR(cidr)
+			ipFamilyMap[ipFamily] = append(ipFamilyMap[ipFamily], cidr)
+		}
+	}
+	return ipFamilyMap
 }
 
-func filterWithCondition(strs []string, expectedCondition bool, conditionFunc func(string) bool) ([]string, []string) {
-	var corrects, incorrects []string
-	for _, str := range strs {
-		if conditionFunc(str) != expectedCondition {
-			incorrects = append(incorrects, str)
-		} else {
-			corrects = append(corrects, str)
-		}
+func getIPFamilyFromIP(ip string) v1.IPFamily {
```

Comment:
So this presents an opportunity for failure and double-parsing.

Would it be better or worse to do it like:

```
func getIPFamilyFromIPString(ipstr string) (v1.IPFamily, error) {
    ip = net.ParseIP(ipstr)
    if ip == nil {
        return "", fmt.Errorf(...)
    }
    if utilnet.IsIPv6(ip) {
		return v1.IPv6Protocol, nil
	}
    return v1.IPv4Protocol, nil
}
```

Then plumb that up to `Map*()` and check for errors there (probably still OK to ignore bad IPs at that level if it is documented).  It just feels weird to not check errors this far down.

?  Or maybe simpler - pass the non-nil net.IP down to getFamily() and call `utilnet.IsIPv6()` instead of ...String ?

Reply:
Sounds good. Will avoid the double parsing. I'll log an error msg for bad IPs/CIDRs and continue with the good ones.